extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -109,7 +107,7 @@ void build(CombinedIndexBuildItem index,
         MetamodelInfo<EntityModel<EntityField>> modelInfo = modelEnhancer.getModelInfo();
         if (modelInfo.hasEntities()) {
             PanacheFieldAccessEnhancer panacheFieldAccessEnhancer = new PanacheFieldAccessEnhancer(modelInfo);
-            for (ClassInfo classInfo : applicationIndex.getIndex().getKnownClasses()) {
+            for (ClassInfo classInfo : index.getIndex().getKnownClasses()) {
```

Comment:
@FroMage `modelClasses` are built from the `CombinedIndexBuildItem` so it should be safe to use it here as well, or?

Reply:
That would probably solve the bug where people access entities outside the root application, but might be costly since we still don't have any way to index users of specific classes.